episodes/basic-gis-with-r-sf.Rmd

```diff
@@ -0,0 +1,245 @@
+bb <- sf::st_bbox(nominatim_polygon)
+x <- opq(bbox = bb) %>%
+   add_osm_feature(key = 'building') %>%
+    osmdata_sf()
+buildings <- x$osm_polygons %>%
+  st_transform(.,crs=28992)
+
+
+summary(buildings$start_date)
+buildings$start_date <- as.numeric(as.character(buildings$start_date))
+
+old_buildings <- buildings %>%
+  filter(start_date <= old)
+
+ ggplot(data = old_buildings) + geom_sf(colour="red")
+ 
+```
+As conservationists, we want to create a zone around historical buildings where building regulation will have special restrictions to preserve historical buildings. 
+
+## Buffers
+
+Let's say this zone should be 100 meters. In GIS terms, we want to create a buffer around polygons. The corresponding function `sf` is `st_buffer`, with 2 arguments: the polygons around which to create buffers, and the radius of the buffer.
+
+
+```{r}
+ distance <- 100 # in meters 
+ 
+buffer_old_buildings <- 
+  st_buffer(x = old_buildings, dist = distance)
+ 
+ ggplot(data = buffer_old_buildings) + geom_sf()
+  
+```
+
+## Union
+
+Now, we have a lot of overlapping buffers. We would rather create a unique conservation zone rather than overlapping ones in that case. So we have to fuse the overlapping buffers into one polygon. This operation is called union and the corresponding function is `st_union`.
+
+```{r}
+ single_old_buffer <- st_union(buffer_old_buildings) %>%
+   st_cast(to = "POLYGON") %>%
+  st_as_sf() 
+
+single_old_buffer<- single_old_buffer %>%
+  mutate("ID"=as.factor(1:nrow(single_old_buffer))) %>%
+  st_transform(.,crs=28992) 
+  
+```
+
+We also use `st_cast()` to explicit the type of the resulting object (*POLYGON* instead of the default *MULTIPOLYGON*), `st_as_sf()` to transform the polygon into an `sf` object.
+
+
+## Centroids
+For the sake of visualisation speed, we would like to represent buildings by a single point (for instance: their geometric centre) rather than their actual footprint. This operation means defining their centroid and the corresponding function is `st_centroid`.
+
+```{r}
+sf::sf_use_s2(FALSE)
+centroids_old <- st_centroid(old_buildings) %>%
+  st_transform(.,crs=28992)  
+
+ggplot() + 
+    geom_sf(data = single_old_buffer, aes(fill=ID)) +
+    geom_sf(data = centroids_old)
+```
+
+## Intersection
+Now what we would like to distinguish conservation areas based on the number of historic buildings they contain. In GIS terms, we would like to know how many centroids each fused buffer polygon contains. This operation means intersecting the layer of polygons with the layer of points and the corresponding function is `st_intersection`.
+
+
+```{r}
+ centroids_buffers <- 
+  st_intersection(centroids_old,single_old_buffer) %>%
+   mutate(n=1)
+
+ centroid_by_buffer <- centroids_buffers %>%
+   group_by(ID) %>%
+   summarise(n = sum(n))
+ single_buffer <- single_old_buffer %>%
+   mutate(n_buildings = centroid_by_buffer$n)
+
```

Comment:
centroid_by_buffer <- centroids_buffers %>%
   group_by(ID)  >> 
the ID which you use to calculate the number of centroids in each buffer corresponds to the single_old_buffer ID, correct? This might not be clear without stating this explicitly during the workshop, so you may want to show or explain that the ID is a factor variable indicating identifying the different polygons in the single_old_buffer layer and after the intersection also in the centroids_buffer layer. By doing so you also partially explain what the intersection function does, because the centroids_buffer layer contains information on both centroids_old and single_old_buffer.